geps/gep-1897.md

```diff
@@ -125,7 +141,326 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+[here](https://github.com/kubernetes-sigs/gateway-api/blob/a33a934af9ec6997b34fd9b00d2ecd13d143e48b/apis/v1alpha2/policy_types.go#L24-L41).
+In an upcoming extension to TargetRef, policy resources _may_ also choose to include `SectionName` and/or `Port` in the
+TargetRef following the same mechanics as `ParentRef`.
+
+TLSConnectionPolicySpec includes `TargetRef` and `TLS` fields.  The `TLS` field is a `TLSConnectionPolicyConfig` and
+contains `TrustedCACertRefs`, `AllowedSANs`, and `TLSVersions` to describe the fields that implementations may cover.
+The names of the fields were chosen to facilitate discussion, but may be substituted without blocking acceptance of the
+content of the API change.
+
+In the API defined here, the `TrustedCACertRefs` is a slice of named config maps containing a single cert. We originally
+proposed to follow the convention established by the
+[CertificateRefs field on Gateway](https://github.com/kubernetes-sigs/gateway-api/blob/18e79909f7310aafc625ba7c862dfcc67b385250/apis/v1beta1/gateway_types.go#L340)
+, but the CertificateRef requires both a tls.key and tls.crt and a `TrustedCACertRef` really only requires a tls.crt.
+`TrustedCACertRefs` contains one or more (up to 64) references to Kubernetes objects that contain PEM-encoded TLS certificates,
+which are used to establish a TLS handshake between the gateway and backend pod.  If empty, then the system trusted
+certificates should be used.  If there are no system trusted certificates or the implementation doesn't define system
+trusted certificates, then a TLS connection must fail when the field is empty.  References to a resource in a
+different namespace are invalid.
+
+TLSConnectionPolicySpec supports the configuration of Subject Alternative Name certificates via the `AllowedSANs` field.
+One or more (up to 64) subject alternative names may be declared.  SAN certificates allow authentication with a single
+certificate that can serve multiple domain
+names that are intended to represent the same endpoint.  For example, these five SAN hosts: example.com, 
+www.example.com, example.net, IPv4 x.x.x.x, and IPv6 x.x.x.x.x.x.x.x all resolve to the same endpoint. If `AllowedSANs`
+is set, then an implementation must validate that the backend pod-provided certificate represents a SAN in its list.
+If empty, then TLS fails.  In the future, we will define more specific ways to handle the case of an empty `AllowedSANs`
+list.
+
+Finally, TLSConnectionPolicy allows the configuration of expected TLS versions via the `TLSVersions` field.
+TLS versions specify one or more (up to 3) TLS versions that the connection may use.  If the `TLSVersions` field is set,
+then the implementation must use the given versions if they are allowed.  If none of the versions are allowed, then the
+TLS connection must fail.  If no TLS versions are specified, then an implementation may choose to apply a self-determined
+default version.
```

Comment:
AFIAK TLS version is based on handshake between client and server, i.e. largest version supported by both. 

A TLSVersion config won't change the range of versions compiled in either client nor server. A backend can simply be compiled with specific version in its options. 

Not sure why the K8S Gateway needs to define alternate API to TLS.

Reply:
TLS version config here is optional, and the implementations may or may not use it to filters failures before they occur, or provide a default that they support if they don't support all versions.